core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -96,12 +96,13 @@ private[deploy] class Worker(
   private val HEARTBEAT_MILLIS = conf.get(WORKER_TIMEOUT) * 1000 / 4
 
   // Model retries to connect to the master, after Hadoop's model.
-  // The first six attempts to reconnect are in shorter intervals (between 5 and 15 seconds)
-  // Afterwards, the next 10 attempts are between 30 and 90 seconds.
+  // The first WORKER_INITIAL_REGISTRATION_RETRIES attempts to reconnect are in shorter intervals
+  // (between 5 and 15 seconds). Afterwards, the next attempts are between 30 and 90 seconds while
+  // The total number of retries are less than or equal to WORKER_MAX_REGISTRATION_RETRIES.
   // A bit of randomness is introduced so that not all of the workers attempt to reconnect at
   // the same time.
-  private val INITIAL_REGISTRATION_RETRIES = 6
-  private val TOTAL_REGISTRATION_RETRIES = INITIAL_REGISTRATION_RETRIES + 10
+  private val INITIAL_REGISTRATION_RETRIES = conf.get(WORKER_INITIAL_REGISTRATION_RETRIES)
+  private val TOTAL_REGISTRATION_RETRIES = conf.get(WORKER_MAX_REGISTRATION_RETRIES)
```

Comment:
Shall we add a checker that ensure TOTAL>INITIAL?

Reply:
Actually, I want to break the relationship because they are conceptually independent.

For example, if maximum = 3 and init = 5, I want `Worker` try 3 times only with the short intervals, @yaooqinn .

Reply:
So if maximum = 3 and init = 5, from the perspective of the actual code logic, the actual effective value of `init` is at most 3?

Reply:
Yes, correct!

Reply:
So, is there a need for a log to clarify this result? Because in the above configuration scenario, init follows the configuration value of maximum rather than its own configuration value.

Reply:
Sure, let me add one.